test/exercises/Mandelbrot/solutions/mandelbrot4-taskpar.chpl

```diff
@@ -17,7 +17,7 @@ config const maxSteps = 50;
 // 
 // The number of tasks we should spawn.
 //
-config const numTasks = here.numCores; // Cores on locale #0.
+config const numTasks = here.numPUs(physical=true); // Cores on locale #0.
```

Comment:
Primarily, I like having a default and I'd be inclined to use it when possible, particularly in codes that are going to be judged for succinctness and elegance.  Now that we have online documentation, I hope people will just read up on the documentation if they want to know the routine's precise behavior (something they'd arguably have to do even with physical=true if they were unfamiliar with it), rather than going to look at locale model code.

But then, the next thing I'm wondering is whether the default should/could be more intelligent.  I.e., if the CHPL_RT_... variable is set to MAX_LOGICAL or MAX_PHYSICAL, could/should this default track that setting.  I.e., it'd suck for the user to run in "logical" mode and have to go switch all their calls to reflect that choice, or introduce a boolean variable of their own to track that choice and have to feed that into every numPU()s call.  And/or should there be some other over-arching boolean switch that chooses between these two modes for both the CHPL_RT setting and the numPUs default?

Then I start thinking about my constant exhortations to not have developers make policy decisions that will affect all programs in 1-1 code reviews (or 1-2 in this case), and think that we should elevate this to a discussion at some level -- core developers, community developers, or users.  Though if the three of us can come up with a proposal that we like to offer up (rather than asking everyone to start from scratch), all the better.